tests-functional/tests/test_wakuext_community_token_permissions.py

```diff
@@ -0,0 +1,286 @@
+        self.owner = backend_new_profile("owner")
+        self.member = backend_new_profile("member")
+        self.non_member = backend_new_profile("non_member")
+        self.fake_address = "0x" + "0" * 40  # Fake address for testing
+
+        # Deploy SNT token for tests that need it
+        self.snt_deployer = SNTDeployer(foundry_client)
+        self.snt_address = self.snt_deployer.snt_contract_address
+        self.controller_address = self.snt_deployer.snt_token_controller_address
+
+        # Create token overrides for wallet service
+        token_overrides = [{"symbol": "SNT", "address": self.snt_address}]
+
+        self.member_with_snt = backend_new_profile("member_with_snt", token_overrides=token_overrides)
+
+        # Fund the member_with_snt with 10 SNT tokens
+        accounts = self.member_with_snt.accounts_service.get_accounts()
+        member_address = accounts[0].get("address") if accounts else self.fake_address
+        token_amount = str(10 * 10**18)  # 10 tokens with 18 decimals
+        generate_cmd = (
+            f"cast send {self.controller_address} 'generateTokens(address,uint256)' "
+            f"{member_address} {token_amount} --rpc-url http://anvil:8545 "
+            f"--private-key {user_1.private_key}"
+        )
+        result = foundry_client.container.exec_run(generate_cmd)
+        logger.debug(f"Generate tokens result for member: exit_code={result.exit_code}, output={result.output.decode()}")
+
+        logger.debug(f"Funded {member_address} with 10 SNT tokens at contract {self.snt_address}")
+
+    def create_token_gated_community(
+        self, owner_backend, permission_type=CommunityTokenPermissionType.BECOME_MEMBER, membership=CommunityPermissionsAccess.AUTO_ACCEPT
+    ):
+        """Helper to create a community with token permissions"""
+        # Create basic community
+        community_resp = owner_backend.wakuext_service.create_community(
+            name=fake.community_name(),
+            description=fake.community_description(),
+            membership=membership,
+        )
+        community_id = community_resp.get("communities", [{}])[0].get("id")
+
+        # Add token permission
+        token_criteria = [
+            {
+                "type": CommunityTokenType.ERC20,
+                "contract_addresses": {31337: "0x1234567890123456789012345678901234567890"},
+                "symbol": "TEST",
+                "amountInWei": "1",  # 1 wei required
+                "decimals": 18,
+            }
+        ]
+
+        permission_resp = owner_backend.wakuext_service.create_community_token_permission(
+            community_id=community_id, permission_type=permission_type, token_criteria=token_criteria
+        )
+
+        return community_id, permission_resp
+
+    @pytest.mark.skip(reason="Pending on issue https://github.com/status-im/status-go/issues/7114")
+    def test_token_gated_community_membership_no_valid_tokens(self):
```

Comment:
I think it's ok to omit, because it's a test in `TestCommunityTokenPermissions`?
```suggestion
    def test_membership_no_valid_tokens(self):
```

Reply:
Fixed at https://github.com/status-im/status-go/pull/7113/commits/64da1b4f4e66b584a3cf8e50992ecb13cfedba1a